### DISCUSSION TITLE
CREATING AN INSTALLER WITH WIX

### DISCUSSION THREAD

**Discussion by usefulservice17** [wixtoolset/issues#7423] [Questions]
Hi everyone,

I am student working on a project to develop an existing installer in C# and I need to understand if It meets my requirements to develop my application:
1. I need the application to be configurable by coding so I can intervene directly, for example to be able to verify the calls to the database
2. verify some prerequisites that the user need to have to run my program, for example having sql server and mondodb
3. To be possible to catch events and to control them and possibly compile them in the next steps
4. leave space to a pre-installation before the actual installation starts

**Comment by robmen** [comment]
I'd recommend starting with the basics by reading the [About the Windows Installer](https://learn.microsoft.com/en-us/windows/win32/msi/about-windows-installer). There is a lot of information there that will give you a foundation for all your questions.
